### DISCUSSION TITLE
Extend StatusBar with GetColor and GetStyle

### DISCUSSION THREAD

**Discussion by Cybrosys** [CommunityToolkit/Maui#1474] [New Feature Discussions]
Hello,

Does anyone see any issues and/or problem with adding the methods `GetColor` and `GetStyle` (`PlatformGetColor` and `PlatformGetStyle`)? 

I find myself needing to retrieve the current values in order to restore the color and/or style to their previous values before leaving a view that changes them.

**Comment by bijington** [comment]
Why do you need to restore them? In my mind the page being restored/returned to should set the color it desires. Does that make sense?

**Reply by Cybrosys** [reply]
Oh yes, and that is also what I was expecting of the `StatusBarBehavior`, but since it is not and some people are advocating for the page who changes thing to restore them, we'd need the Get-methods in order to support that. But this is what we're discussing in #1194 and #1471 . 

But looking purely on adding the extra methods, do you see any issues and/or problems with adding them? It would enable the `StatusBarBehavior` to expose the current color and style after being attached instead of just having its current default values.

**Reply by pictos** [reply]
@Cybrosys why would we need the methods, if the Behavior exposed those values as properties?

**Comment by CliffAgius** [comment]
Closed and answered sort of. 😉
